pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -56,27 +56,30 @@ type ClientConfig interface {
 	Namespace() (string, bool, error)
 }
 
+type PersistAuthProviderConfigForUser func(user string) restclient.AuthProviderConfigPersister
+
 // DirectClientConfig is a ClientConfig interface that is backed by a clientcmdapi.Config, options overrides, and an optional fallbackReader for auth information
 type DirectClientConfig struct {
-	config         clientcmdapi.Config
-	contextName    string
-	overrides      *ConfigOverrides
-	fallbackReader io.Reader
+	config                    clientcmdapi.Config
+	contextName               string
+	overrides                 *ConfigOverrides
+	fallbackReader            io.Reader
+	persistAuthProviderConfig PersistAuthProviderConfigForUser
```

Comment:
Any caller that has the ability to persist should be able to handle providing a persister without too much trouble. A caller that doesn't have a notion of persisting the config may not be able to provide a persister, in which case the plugin should handle a nil persister gracefully.

Is that a reasonable assertion for now?